openeo_udp/eurac_pv_farm_detection/udf_eurac_pvfarm_onnx.py

```diff
@@ -0,0 +1,189 @@
+import functools
+import logging
+import os
+import sys
+import zipfile
+from typing import Dict
+
+import numpy as np
+import requests
+import xarray as xr
+
+
+def _setup_logging():
+    logging.basicConfig(level=logging.INFO)
+    logger = logging.getLogger(__name__)
+    return logger
+
+
+# TODO move standard code to UDF repo
+
+# Fixed directories for dependencies and model files
+DEPENDENCIES_DIR = "onnx_dependencies"
+MODEL_DIR = "model_files"
+
+
+def download_file(url, path):
+    """
+    Downloads a file from the given URL to the specified path.
+    """
+    response = requests.get(url, stream=True)
+    with open(path, "wb") as file:
+        file.write(response.content)
+
+
+def extract_zip(zip_path, extract_to):
+    """
+    Extracts a zip file from zip_path to the specified extract_to directory.
+    """
+    with zipfile.ZipFile(zip_path, "r") as zip_ref:
+        zip_ref.extractall(extract_to)
+    os.remove(zip_path)  # Clean up the zip file after extraction
+
+
+def add_directory_to_sys_path(directory):
+    """
+    Adds a directory to the Python sys.path if it's not already present.
+    """
+    if directory not in sys.path:
+        sys.path.append(directory)
+
+
+def setup_model_and_dependencies(model_url, dependencies_url):
+    """
+    Main function to set up the model and dependencies by downloading, extracting,
+    and adding necessary directories to sys.path.
+    """
+
+    logger = _setup_logging()
+    logger.info("Create directories")
+    # Ensure base directories exist
+    os.makedirs(DEPENDENCIES_DIR, exist_ok=True)
+    os.makedirs(MODEL_DIR, exist_ok=True)
+
+    # Download and extract dependencies if not already present
+    if not os.listdir(DEPENDENCIES_DIR):
+        logger.info("Extract dependencies")
+        zip_path = os.path.join(DEPENDENCIES_DIR, "temp.zip")
+        download_file(dependencies_url, zip_path)
+        extract_zip(zip_path, DEPENDENCIES_DIR)
```

Comment:
This is probably out of scope of this ticket, but I think there is a race condition problem here: because of non-zero time between the `os.listdir(DEPENDENCIES_DIR)` check and actually writing the download to file, there could be multiple executors doing the same download or unzip, potentially causing corruption

Reply:
Indeed.

would it help to cache the extraction functionality?

Reply:
no, caching (I assume you mean in-process caching) won't help because the problem lies in concurrency between different executors (different processes possibly on different hosts).

A solution would be to work with a lock file to ensure ownership before starting the download. But as mentioned, this is probably out of scope of this ticket, and probably better handled when incorporating this helper functionality in the python client